File: 70. Climbing Stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    if (n == 1 || n == 2)
        return n;
    let pre1 = 1, pre2 = 2;
    for (let i = 3; i <= n; i++)
    {
        let temp = pre2;
        pre2 = pre1 + pre2;
        pre1 = temp;
    }
    return pre2;
};

File: 70. Climbing Stairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 1 or n == 2:
            return n
        pre1 = 1
        pre2 = 2
        for i in xrange(3, n+1):
            temp = pre2
            pre2 = pre1 + pre2
            pre1 = temp
        return pre2
